File: leetcode/javascript/338.js
// Leetcode 338 - Counting Bits
// Dec. 1, 2023

/**
 * @param {number} n
 * @return {number[]}
 */
var countBits = function(n) {
    let ans = Array(n+1).fill(0);

    for (let i = 1; i < n + 1; i++) ans[i] = ans[parseInt(i / 2)] + (i % 2);

    return ans;
};

File: leetcode/c/338.c
// Leetcode 338 - Counting Bits
// Dec. 1, 2023

int* countBits(int n, int* returnSize) {
    *returnSize = n + 1;
    int* ans = (int*)malloc(sizeof(int) * *returnSize);
    ans[0] = 0;

    for (int i = 1; i < *returnSize; i++) ans[i] = ans[i/2] + (i % 2);

    return ans;
}